services/galley/src/Galley/Data/Queries.hs

```diff
@@ -14,26 +14,101 @@ import Data.Monoid
 import Data.Text (Text)
 import Galley.Types hiding (Conversation)
 import Galley.Types.Bot
+import Galley.Types.Teams
 
 import qualified Data.Text.Lazy as LT
 
+-- Teams --------------------------------------------------------------------
+
+selectTeam :: PrepQuery R (Identity TeamId) (UserId, Text, Text, Maybe Text, Bool)
+selectTeam = "select creator, name, icon, icon_key, deleted from team where team = ?"
+
+selectTeamConv :: PrepQuery R (TeamId, ConvId) (Identity Bool)
+selectTeamConv = "select managed from team_conv where team = ? and conv = ?"
+
+selectTeamConvs :: PrepQuery R (Identity TeamId) (ConvId, Bool)
+selectTeamConvs = "select conv, managed from team_conv where team = ? order by conv"
+
+selectTeamMember :: PrepQuery R (TeamId, UserId) (Identity Permissions)
+selectTeamMember = "select perms from team_member where team = ? and user = ?"
+
+selectTeamMembers :: PrepQuery R (Identity TeamId) (UserId, Permissions)
+selectTeamMembers = "select user, perms from team_member where team = ? order by user"
+
+selectUserTeams :: PrepQuery R (Identity UserId) (Identity TeamId)
+selectUserTeams = "select team from user_team where user = ? order by team"
+
+selectUserTeamsIn :: PrepQuery R (UserId, [TeamId]) (Identity TeamId)
+selectUserTeamsIn = "select team from user_team where user = ? and team in ? order by team"
+
+selectUserTeamsFrom :: PrepQuery R (UserId, TeamId) (Identity TeamId)
+selectUserTeamsFrom = "select team from user_team where user = ? and team > ? order by team"
+
+insertTeam :: PrepQuery W (TeamId, UserId, Text, Text, Maybe Text) ()
+insertTeam = "insert into team (team, creator, name, icon, icon_key, deleted) values (?, ?, ?, ?, ?, false)"
+
+insertTeamConv :: PrepQuery W (TeamId, ConvId, Bool) ()
+insertTeamConv = "insert into team_conv (team, conv, managed) values (?, ?, ?)"
+
+deleteTeamConv :: PrepQuery W (TeamId, ConvId) ()
+deleteTeamConv = "delete from team_conv where team = ? and conv = ?"
+
+insertTeamMember :: PrepQuery W (TeamId, UserId, Permissions) ()
+insertTeamMember = "insert into team_member (team, user, perms) values (?, ?, ?)"
+
+deleteTeamMember :: PrepQuery W (TeamId, UserId) ()
+deleteTeamMember = "delete from team_member where team = ? and user = ?"
+
+updatePermissions :: PrepQuery W (Permissions, TeamId, UserId) ()
+updatePermissions = "update team_member set perms = ? where team = ? and user = ?"
+
+insertUserTeam :: PrepQuery W (UserId, TeamId) ()
+insertUserTeam = "insert into user_team (user, team) values (?, ?)"
+
+deleteUserTeam :: PrepQuery W (UserId, TeamId) ()
+deleteUserTeam = "delete from user_team where user = ? and team = ?"
+
+markTeamDeleted :: PrepQuery W (Identity TeamId) ()
+markTeamDeleted = "update team set deleted = true where team = ?"
+
+deleteTeam :: PrepQuery W (Identity TeamId) ()
+deleteTeam = "delete from team using timestamp 32503680000000000 where team = ?"
```

Comment:
Trying to educate myself here, looking at the [docs](http://docs.datastax.com/en/cql/3.3/cql/cql_reference/cqlDelete.html) this would mean deleting values older than this timestamp. What is the benefit setting it here versus not setting any timestamp?